src/RefractAST.cc

```diff
@@ -194,47 +194,75 @@ namespace drafter
 
     static refract::IElement* MsonElementToRefract(const mson::Element& mse);
 
-    // FIXME: check against original behavioration
-    template <typename T, typename V = typename T::ValueType>
+    RefractElements MsonElementsToRefract(const mson::Elements& elements) {
+        RefractElements result;
+        std::transform(elements.begin(), elements.end(), std::back_inserter(result), MsonElementToRefract);
+        return result;
+    }
+
+    struct TypeSectionData {
+        RefractElements defaults;
+        RefractElements samples;
+        std::vector<std::string> descriptions;
+    };
+
+    template <typename T>
     struct ExtractTypeSection
     { // This will handle primitive elements
```

Comment:
The previous PR had a comment on this line saying that this comment needs to be removed. I haven't looked into it yet but wanted to mention it.